src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const employeesContainer = document.querySelector('ul');
+const employees = [...document.querySelectorAll('[data-salary]')];
+const employeesList = employees.map((employee) => {
+  return {
+    name: employee.innerText,
+    position: employee.dataset.position,
+    salary: +employee.dataset.salary.replace(/[$,]/g, ''),
+    age: employee.dataset.age,
+  };
+});
+
+const sortEmployees = employeesList.sort((a, b) => b.salary - a.salary);
+const onStartOrderedList = () => {
+  employeesContainer.innerHTML = '';
+
+  sortEmployees.forEach((el) => {
+    employeesContainer.innerHTML += `
+      <li
+        data-position=${el.position}
+        data-salary=${el.salary}
+        data-age=${el.age}
+      >
+        ${el.name}
+      </li>
+    `;
+  });
+};
```

Comment:
Directly manipulating innerHTML in a loop can lead to performance issues, especially with larger lists. Each assignment to 'innerHTML' causes the browser to re-parse the entire HTML structure, which can be inefficient. Instead, consider creating the elements using document.createElement and then appending them to the container using appendChild or using insertAdjacentHTML to insert the HTML string once after the loop.